src/sgame/sg_bot_parse.cpp

```diff
@@ -142,9 +142,38 @@ static AIValue_t haveUpgrade( gentity_t *self, const AIValue_t *params )
 	return AIBoxInt( !BG_UpgradeIsActive( upgrade, self->client->ps.stats ) && BG_InventoryContainsUpgrade( upgrade, self->client->ps.stats ) );
 }
 
-static AIValue_t percentAmmo( gentity_t *self, const AIValue_t* )
+// This function returns the ratio of 2 ammo values for a
+// specific weapon. Only used to reduce duplicated code.
+// TODO: make this a weapon_t or weaponAttributes_t method?
+static AIValue_t ratio( int current, int max, weapon_t wp )
```

Comment:
Function name should begin with capital letter

If it's an infinite ammo weapon, `max` should be 0 right? So maybe `wp` is not needed.

Reply:
barbs are infinite, yet they have a max value.

Reply:
Barbs are not infinite, you can only have 3. And the maxAmmo value is 3. So everything works out by just ignoring the infiniteAmmo field.

Reply:
barbs are infinite, but they have a max amount. This is obviously super weird, but this is that kind of super weird things which makes the code so hard to get right or generic enough to prevent code duplication everywhere.
I do not remember all the details on that, there's stuff to clean obviously. The whole equipment/class/upgrade system is in a very bad state there.

See [this](https://github.com/UnvanquishedAssets/unvanquished_src.dpkdir/blob/08a511f95cfc7bff735109127d27fcdfc5ba88f9/configs/weapon/level3upg.attr.cfg#L11) as a proof I'm not spouting random noise through keyboard.

Reply:
Yes I understand that. I'm saying ignore infiniteAmmo and base the decision only on whether maxAmmo is 0 - give a resupply score of 0 whenever there is 0 maxAmmo. Hypothetically speaking, if there were a gun that has 0 ammo and non-infinite ammo (thus unfirable), you would also want to return 0, and need this to avoid dividing by 0.

Reply:
> if there were a gun that has 0 ammo and non-infinite ammo (thus unfirable)

Good point that it's technically possible to get that... I'll add a Log::Warn() too.